### PR TITLE
feat(collections): add dropLastWhile

### PR DESCRIPTION
**Pull request by grian32**

### CODE REVIEW

File: collections/README.md [hunk 1]
```diff
@@ -586,3 +586,24 @@ const personWithMinAge = minBy(people, (i) => i.age);
 
 assertEquals(personWithMinAge, { name: "John", age: 23 });
 ```
+
+### dropLastWhile
+
+Returns all elements in the given collection until the last element that does
```
Comment (getspooky) [inline collections/README.md:592]:
```suggestion
Drop all elements in the given collection until the last element that does
```

File: collections/README.md [hunk 2]
```diff
@@ -586,3 +586,24 @@ const personWithMinAge = minBy(people, (i) => i.age);
...
+
+Example:
+
+```ts
+import { dropLastWhile } from "https://deno.land/std@$STD_VERSION/collections/mod.ts";
+import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
+
+const values = [22, 30, 44];
+
+const notFourtyFour = dropLastWhile(values, (i) => i != 44);
```
Comment (getspooky) [inline collections/README.md:603]:
```suggestion
const dropLastWhileNumbers = dropLastWhile(numbers, (i) => i != 44);
```

File: collections/README.md (hunk 1 above)
Comment (getspooky) [inline collections/README.md:592]:
```suggestion
Drop all elements in the given collection until the last element that does
```

File: collections/README.md [hunk 3]
```diff
@@ -586,3 +586,24 @@ const personWithMinAge = minBy(people, (i) => i.age);
...
+Returns all elements in the given collection until the last element that does
+not match the given predicate
+
+Example:
+
+```ts
+import { dropLastWhile } from "https://deno.land/std@$STD_VERSION/collections/mod.ts";
+import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
+
+const values = [22, 30, 44];
```
Comment (getspooky) [inline collections/README.md:601]:
```suggestion
const numbers = [22, 30, 44];
```

File: collections/drop_last_while.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * const values = [22, 30, 44];
+ *
+ * const notFourtyFour = dropLastWhile(values, i => i != 44);
+ *
+ * assertEquals(
+ *   notFourtyFour,
+ *   [22, 30]
+ * );
+ * ```
```
Comment (getspooky) [inline collections/drop_last_while.ts:17]:
```suggestion
 * Drop all elements in the given collection until the last element that does not match the given predicate
 *
 * Example:
 * ```ts
 * import { dropLastWhile } from "./drop_last_while.ts";
 * import { assertEquals } from "../testing/asserts.ts";
 *
 * const numbers = [22, 30, 44];
 *
 * const dropLastWhileNumbers = dropLastWhile(numbers, i => i != 44);
 *
 * assertEquals(
 *   dropLastWhileNumbers,
 *   [22, 30]
 * );
 * ```
```

File: collections/README.md [hunk 5]
```diff
@@ -586,3 +586,24 @@ const personWithMinAge = minBy(people, (i) => i.age);
...
+
+const notFourtyFour = dropLastWhile(values, (i) => i != 44);
+
+assertEquals(
+  notFourtyFour,
```
Comment (getspooky) [inline collections/README.md:606]:
```suggestion
dropLastWhileNumbers,
```

**Review by getspooky — CHANGES_REQUESTED** [review]

File: collections/drop_last_while.ts (hunk 4 above)
Comment (grian32) [reply collections/drop_last_while.ts]: I disagree, `notFourtyFour` is more descriptive and generally makes more sense than `dropLastWhileNumbers`

File: collections/README.md (hunk 5 above)
Comment (grian32) [reply collections/README.md:606]: same as above

**Review by getspooky — APPROVED** [review]

**Comment by getspooky** [comment]
LGTM, @kt3k

**Comment by LionC** [comment]
> LGTM, @kt3k

For what it's worth, I agree :-)

**Review by kt3k — APPROVED** [review]
@grian32 LGTM. Thank you for your contribution!

@getspooky @LionC Thank you for your reviews

**Pull request merged**